otherlibs/stdune-unstable/path.ml

```diff
@@ -302,6 +302,8 @@ end = struct
       User_error.raise ?loc:error_loc
         [ Pp.textf "path outside the workspace: %s from %s" path (to_string t) ]
 
+  (* Check whether a path is in canonical form: no '.' or '..' components, no
+     repeated '/' components, and no backslashes '\\' (on Windows only) *)
```

Comment:
And the function also always returns false if the path ends with a `/` (even when path is the root `/` and is, in fact, already canonical...)